File: Pokemon_Game/Pokemon.py
from abc import ABC, abstractmethod

class Pokemon(ABC):
    def __init__(self, nickname, strength, weakness):
        self.__nickname = nickname
        self.__strength = strength
        self.__weakness = weakness

    @abstractmethod
    def battle_cry(self):
        pass

    @property
    def nickname(self):
        return self.__nickname

    @property
    def strength(self):
        return self.__strength

    @property
    def weakness(self):
        return self.__weakness

File: PokemonGame_CSharp/Pokemon.cs
public enum ElementType
{
    Fire,
    Water,
    Grass
}

public abstract class Pokemon
{
    public string Nickname { get; }
    public ElementType Strength { get; }
    public ElementType Weakness { get; }

    protected Pokemon(string nickname, ElementType strength, ElementType weakness)
    {
        Nickname = nickname;
        Strength = strength;
        Weakness = weakness;
    }

    public abstract void BattleCry();
}

public class Charmander : Pokemon
{
    public Charmander(string nickname) : base(nickname, ElementType.Fire, ElementType.Water)
    {
    }

    public override void BattleCry()
    {
        Console.WriteLine($"{Nickname}!");
    }
}

public class Squirtle : Pokemon
{
    public Squirtle(string nickname) : base(nickname, ElementType.Water, ElementType.Grass)
    {
    }

    public override void BattleCry()
    {
        Console.WriteLine($"{Nickname}!");
    }
}

public class Bulbasaur : Pokemon
{
    public Bulbasaur(string nickname) : base(nickname, ElementType.Grass, ElementType.Fire)
    {
    }

    public override void BattleCry()
    {
        Console.WriteLine($"{Nickname}!");
    }
}
